hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionInfo.java

```diff
@@ -346,15 +346,23 @@ static TableName getTable(final byte [] regionName) {
     return parseRegionName(regionName)[1];
   }
 
-  @InterfaceAudience.Private
-  static boolean isEncodedRegionName(byte[] regionName) throws IOException {
+  /**
+   * Figure if the passed bytes represent an encoded region name or not.
+   * @param regionName A Region name either encoded or not.
+   * @return True if <code>regionName</code> represents an encoded name.
+   */
+  @InterfaceAudience.Private // For use by internals only.
+  public static boolean isEncodedRegionName(byte[] regionName) throws IOException {
     try {
+      // TODO: This is extreme... a full parse and an exception if encoded -- and
+      // even then, we say it encoded just because it NOT encoded. Can we
+      // do something less expensive figuring if encoded?
```

Comment:
How expensive is checking all the encoded region names we know about?

Reply:
Or! What about keeping a cache of the last N already checked names? 1m?

Reply:
I should have added 'relatively'. This call happens rarely in scheme of things. Can address its expense when shows in a profiler.

Reply:
softening the TODO language would definitely help then 😸

Reply:
You are right. Let me fix.